.travis.yml

```diff
@@ -1,52 +1,124 @@
-language: c
+# Configuration file for TravisCI
+
+# Use the container builds so we don't need sudo priviledges
+sudo: false
 
-sudo: required
+language: c
 
+# Only build pushes to select branches and tags. This avoids the double
+# builds than happen when working on a branch instead of a fork.
 branches:
     only:
         - master
         - 5.4
         # Regex to build tagged commits with version numbers
         - /\d+\.\d+(\.\d+)?(\S*)?$/
 
-os:
-    - linux
-    #- osx
+# Set the Ubuntu version for the Linux builds
+dist: xenial
+
+# Dependencies for the Linux and OSX builds
+addons:
+    apt:
+        packages:
+            - cmake
+            - build-essential
+            - libcurl4-gnutls-dev
+            - libnetcdf-dev
+            - libgdal1-dev
+            - libfftw3-dev
+            - libpcre3-dev
+            - liblapack-dev
+            - ghostscript
+            - curl
+            - graphicsmagick
+            - python
+            - python-pip
+    homebrew:
+        packages:
+            - cmake
+            - curl
+            - netcdf
+            - gdal
+            - fftw
+            - pcre2
+            - ghostscript
+            - curl
+            - graphicsmagick
+            - python
 
+# Define environment variables and set build actions
 env:
     global:
         - INSTALLDIR="$HOME/gmt-install-dir"
         - COASTLINEDIR="$HOME/gmt-install-dir/coast"
         - PATH="$INSTALLDIR/bin:$PATH"
         - LD_LIBRARY_PATH="$INSTALLDIR/lib:$LD_LIBRARY_PATH"
-        - if [ "$TRAVIS_EVENT_TYPE" == "cron" ]; then
-            export PYTHON=3.6
-            export CONDA_INSTALL_EXTRA="sphinx"
-            export COVERAGE=true
-          fi
+        - COVERAGE=true
```

Comment:
The environment variable COVERAGE seems useless, since we always need to run the tests to generate coverage reports.

Reply:
Yeah, I agree. I'll remove it and use the `TEST` variable instead.